scenes/music_puzzle/music_puzzle.gd

```diff
@@ -0,0 +1,61 @@
+extends Node2D
+
+@export var xylophone: MusicalRockXylophone
+
+## Melodies expressed with the letters ABCDEFG.
+@export var melodies: Array[String]
+
+## A fire corresponding to each melody, ignited when the correct melody is played (in order).
+@export var fires: Array[Bonfire]
+
+signal solved
+
+var _current_melody := 0
+var _position := 0
+
+
+func _ready() -> void:
+	xylophone.note_played.connect(_on_note_played)
+
+
+func _debug(fmt: String, args: Array = []) -> void:
+	if false:
```

Comment:
I think this false could be turned into an @export variable, so we can flip that switch while the game is running in the editor in case we want to check the debug lines.
```suggestion
	if false:
```